cibyl/plugins/openstack/printers/serialized.py

```diff
@@ -0,0 +1,141 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+from abc import ABC
+
+from overrides import overrides
+
+from cibyl.outputs.cli.printer import JSONPrinter, SerializedPrinter
+from cibyl.plugins.openstack import Deployment
+from cibyl.plugins.openstack.container import Container
+from cibyl.plugins.openstack.node import Node
+from cibyl.plugins.openstack.package import Package
+from cibyl.plugins.openstack.printers import OSPrinter
+from cibyl.plugins.openstack.service import Service
+
+
+class OSSerializedPrinter(OSPrinter, SerializedPrinter, ABC):
+    """Provides a machine-readable representation of the plugin's models for
+    easy readability from a machine.
+    """
+
+    @overrides
+    def print_deployment(self, deployment: Deployment) -> str:
+        network = deployment.network.value
+        storage = deployment.storage.value
+        ironic = deployment.ironic.value
+
+        result = {
+            'release': deployment.release.value,
+            'infra_type': deployment.infra_type.value,
+            'topology': deployment.topology.value,
+            'network': {
+                'ip_version': network.ip_version.value,
+                'network_backend': network.network_backend.value,
+                'ml2_driver': network.ml2_driver.value,
+                'security_group': network.security_group.value,
+                "dvr": network.dvr.value,
+                'tls_everywhere': network.tls_everywhere.value
+            },
+            'storage': {
+                'cinder_backend': storage.cinder_backend.value
+            },
+            'ironic': {
+                'ironic_inspector': ironic.ironic_inspector.value,
+                'cleaning_network': ironic.cleaning_network.value
+            },
+            'overcloud_templates': list(deployment.overcloud_templates.value),
```

Comment:
`overcloud_templates` could be `None` and this line would crash. I got the error running `cibyl -vv -f json --config jenkins.yaml query --jobs DFG-network-openstack-designate-17.0_director-rhel-virthost-3cont_2comp-ipv4-geneve --last-build --ip-version --cinder-backend` against downstream Jenkins